src/app/courses/courses.component.html

```diff
@@ -1,5 +1,5 @@
 <mat-toolbar>
-  <button mat-icon-button (click)="goBack()"><mat-icon>arrow_back</mat-icon></button>
+  <button mat-icon-button (click)="planetMatTableService.goBack(parent)"><mat-icon>arrow_back</mat-icon></button>
```

Comment:
goback is not a part of mattable. It is used across all application to route back to relevant upper hierarchy page.